pynecone/.templates/web/utils/state.js

```diff
@@ -207,6 +208,11 @@ export const connect = async (
   // Once the socket is open, hydrate the page.
   socket.current.on("connect", () => {
     updateState(state, setState, result, setResult, router, socket.current);
+    setNotConnected(false)
+  });
+
+  socket.current.on('connect_error', (error) => {
```

Comment:
I think it would be nice to stash this `error` in the component state and then export a `BaseVar` that references the state value. Then the default connection error banner could render as a `pc.cond` on that var (and it could optionally render the actual error message).

With this approach, the page template wouldn't need special handling to render the error component, since the `connectionError` could be referenced in the compiled page.

Reply:
Great idea, we will add this in a follow-up